### DISCUSSION TITLE
Request for publishing standalone binaries

### DISCUSSION THREAD

**Discussion by brainchild0** [dani-garcia/vaultwarden#3765] [Ideas]
The assets published through GitHub for each release are simply source archives. The only published executable system is through container images. Documentation details a process for manually extracting binaries from the image, but the process is inconvenient, and requires that particular tools and resources be available locally for the end user.

Ideally, a containerized build process would be integrated with revision tracking, as through GitHub actions, for building releases, following a process that is closely relating to building the container images.

However, a solution requiring less complex changes, from the current situation, may be simply automating the process of extracting binaries from images. Such a process would substitute for directly building the targets as standalone components, leveraging the build process having already occurred in the image building process.

**Comment by BlackDex** [comment]
This is something on my personal roadmap. And I want to remove the binaries from the test builds on the Actions page also.
But I need to overhaul the build process for this because we currently use a build script which also doesn't play nice with caching for example.

There are multiple items on which I'm currently working so this is not one of them right now, but it is on my list at least.

**Reply by brainchild0** [reply]
I am not understanding the reference to the problems related to caching during the build process, but automation of the process for extracting binaries from images may be a more lightweight solution, until a better one is feasible.

The result would be simply that users could obtain the identical assets as through the recommended process, but through a simple download rather than manual operation on the local system.

**Reply by BlackDex** [reply]
I understand what the outcome would be.

I just wanted to provide a bit more detailed into what i tend to do, and as mentioned in my previous post, I'm currently working on other items when I have time.

So, it is on my list, if someone else feels obligated to help and improve the process feel free.
